### ISSUE TITLE
Make deploy preview links point to updated pages

### ISSUE THREAD

**Issue by prasadkatti** [kubernetes/website#20814]
**This is a Feature Request**

**What would you like to be added**
In most cases, the PRs in this repo will update one specific page. For such cases, can we make the deploy preview link point specifically to the changed page?

For e.g., for the PR https://github.com/kubernetes/website/pull/20812, the deploy preview link that is created by the Netlify bot is -
https://deploy-preview-20812--kubernetes-io-master-staging.netlify.app/
Instead, if it were -
https://deploy-preview-20812--kubernetes-io-master-staging.netlify.app/docs/reference/using-api/api-concepts/
that would make the reviewer's life easier.

**Why is this needed**
Would make reviewer's job easier and faster

**Comments**

This will require changes to the netlify bot. Not sure how configurable the bot is.

**Comment by prasadkatti** [comment]
Using `git diff-tree` and PR number, it should be fairly easy to generate a direct link to the changed page, imo.
```
$ git diff-tree --no-commit-id --name-only -r HEAD
content/en/docs/reference/using-api/api-concepts.md
```

**Comment by sftim** [comment]
This is a great idea.

I also agree that changes to Netlify behavior are not necessarily straightforward.

**Comment by kbhawkey** [comment]
Yes, this has been on my wish list for some time.

**Comment by prasadkatti** [comment]
So I looked at Netlify docs and it doesn't seem like we can make their bot do what we want it to do. Is creating a separate bot to do this an option here? If yes, I would like to work on it. But want to be clear what the right approach is before starting.

**Comment by tengqm** [comment]
/remove-lifecycle stale
Still something desirable.

**Comment by sftim** [comment]
Duplicate of #4657
